RELEASE.md

```diff
@@ -22,11 +22,20 @@ A release for the bpfd project is comprised of the following major components:
   - `quay.io/bpfd/bpfd-operator`
```

Comment:
Should there be a link to this doc in the Development section of the docs Website?